walkthroughs/ec2/ec2-cluster.yaml

```diff
@@ -0,0 +1,177 @@
+    Type: String
+
+  InstanceType:
+    Description: Which instance type should we use to build the EC2 cluster?
+    Type: String
+    Default: c4.large
+
+  EC2AMI:
+    Description: The Amazon Machine Image ID used for the cluster, leave it as the default value to get the latest AMI
+    Type: AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>
+    Default: /aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2
+#    Default: /aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id
+#    Default: ami-08e689a423405d7f6 amzn2-ami-hvm-2.0.20190313-x86_64-ebs
+
+  KeyName:
+    Type: AWS::EC2::KeyPair::KeyName
+    Description: The name of an existing EC2 KeyPair that will be used to access the underlying hosts in the cluster. If none appear here you need to create and download an SSH key.
+
+  ProxyRouterManagerImage:
+    Type: String
+    Description: Sidecar router manager that sets up networking for transparent proxy
+    Default: "subfuzion/aws-appmesh-proxy-route-manager:latest"
+
+  EnvoyImage:
+    Type: String
+    Description: Envoy image to use
+
+  AppMeshXdsEndpoint:
+    Type: String
+    Description: App Mesh XDS Endpoint Override
+    Default: ""
+
+  EnvoyLogLevel:
+    Type: String
+    Default: debug
+
+  ColorTellerImage:
+    Type: String
+    Description: Color Teller app image
+    
+  AppMeshMeshName:
+    Type: String
+    Description: Name of mesh
+
+  AppMeshEgressIgnoredIpCsv:
+    Type: String
+    Default: 169.254.170.2,169.254.169.254
+
+  ECSServicesDomain:
+    Type: String
+    Description: DNS namespace used by services e.g. default.svc.cluster.local
+
+Resources:
+
+  EC2SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Allow connections on specified ports
+      VpcId:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:VPC"
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+        - IpProtocol: tcp
+          FromPort: 9080
+          ToPort: 9080
+          CidrIp: 0.0.0.0/0
+
+  # EC2 instance for running Docker containers
+  EC2Instance:
+    Type: AWS::EC2::Instance
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT10M
+    Properties:
+      ImageId: !Ref EC2AMI
+      InstanceType: !Ref InstanceType
+      SecurityGroupIds:
+        - !Ref EC2SecurityGroup
+      SubnetId:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:PublicSubnet1"
+      KeyName: !Ref KeyName
+      Tags:
+        - Key: Name
+          Value: !Sub ${EnvironmentName}-ec2-instance
+      UserData:
+        Fn::Base64:
+          !Sub |
+            #!/bin/bash -x
+
+            # CloudFormation support - use to signal cloudformation when finished
+            yum install -y aws-cfn-bootstrap
+
+            # golang support - use to build colorteller app
+            amazon-linux-extras install -y golang1.11
+            mkdir -p ~/go ~/go/bin ~/go/pkg ~/go/src/github.com/aws
```

Comment:
how big is the output of the colorteller app? maybe we should base64 it and inject it into this script =p

Reply:
lol